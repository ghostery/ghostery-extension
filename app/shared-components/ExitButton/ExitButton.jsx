/**
 * Exit Button Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

/**
 * A Functional React component for a Exit Button
 * @return {JSX} JSX for rendering a Exit Button
 * @memberof SharedComponents
 */
const ExitButton = (props) => {
	const {
		hrefExit,
		textExit,
	} = props;

	return (
		<NavLink to={hrefExit} className="ExitButton__exit flex-container align-middle">
			{textExit && (
				<span className="ExitButton__exitText">{textExit}</span>
			)}
			<span className="ExitButton__exitIcon" />
		</NavLink>
	);
};

// PropTypes ensure we pass required props of the correct type
ExitButton.propTypes = {
	hrefExit: PropTypes.string.isRequired,
	textExit: PropTypes.oneOfType([
		PropTypes.bool,
		PropTypes.string,
	]).isRequired,
};

export default ExitButton;
