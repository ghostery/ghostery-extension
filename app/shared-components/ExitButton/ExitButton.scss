/**
 * Exit Button Sass
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

// Exit Button
.ExitButton__exit {
	position: absolute;
	top: 0;
	right: 0;
	margin: 18px 33px;
	padding: 11px 13px;
	border-radius: 3px;
}
.ExitButton__exit:hover {
	background-color: #efefef;
}
.ExitButton__exitText {
	font-size: 11px;
	color: #4a4a4a;
	margin-right: 15px;
}
.ExitButton__exitIcon {
	height: 15px;
	width: 15px;
	margin: 0 auto;
	background-repeat: no-repeat;
	background-position: center center;
	background-image: buildIconX(#4a4a4a);
}
