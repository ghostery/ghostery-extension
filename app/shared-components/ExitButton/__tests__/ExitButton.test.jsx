/**
 * Exit Button Test Component
 *
 * Ghostery Browser Extension
 * https://www.ghostery.com/
 *
 * Copyright 2018 Ghostery, Inc. All rights reserved.
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router';
import ExitButton from '../ExitButton';

describe('app/shared-components/ExitButton', () => {
	describe('Snapshot tests with react-test-renderer', () => {
		test('exit button is rendered correctly', () => {
			const initialState = {
				hrefExit: '/',
				textExit: 'Exit',
			};
			const component = renderer.create(
				<MemoryRouter>
					<ExitButton {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('More Snapshot tests with react-test-renderer, but for edge cases', () => {
		test('edge case where text is empty string', () => {
			const initialState = {
				hrefExit: '/test',
				textExit: '',
			};
			const component = renderer.create(
				<MemoryRouter>
					<ExitButton {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});

		test('edge case where text is false', () => {
			const initialState = {
				hrefExit: '/example',
				textExit: false,
			};
			const component = renderer.create(
				<MemoryRouter>
					<ExitButton {...initialState} />
				</MemoryRouter>
			).toJSON();
			expect(component).toMatchSnapshot();
		});
	});

	describe('Shallow snapshot tests rendered with Enzyme', () => {
		test('the happy path of the component', () => {
			const initialState = {
				hrefExit: '/',
				textExit: 'Exit',
			};

			const component = shallow(<ExitButton {...initialState} />);
			expect(component.find('.ExitButton__exit').length).toBe(1);
			expect(component.find('.ExitButton__exitText').length).toBe(1);
			expect(component.find('.ExitButton__exitIcon').length).toBe(1);
		});

		test('the edge cases of the component', () => {
			const initialState = {
				hrefExit: '/',
				textExit: '',
			};

			const component = shallow(<ExitButton {...initialState} />);
			expect(component.find('.ExitButton__exit').length).toBe(1);
			expect(component.find('.ExitButton__exitText').length).toBe(0);
			expect(component.find('.ExitButton__exitIcon').length).toBe(1);

			// Same result when textExit is boolean instead of empty string
			component.setProps({ textExit: false });
			expect(component.find('.ExitButton__exit').length).toBe(1);
			expect(component.find('.ExitButton__exitText').length).toBe(0);
			expect(component.find('.ExitButton__exitIcon').length).toBe(1);
		});
	});
});
