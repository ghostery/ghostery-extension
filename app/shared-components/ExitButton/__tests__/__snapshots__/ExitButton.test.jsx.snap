// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/shared-components/ExitButton More Snapshot tests with react-test-renderer, but for edge cases edge case where text is empty string 1`] = `
<a
  aria-current={null}
  className="ExitButton__exit flex-container align-middle"
  href="/test"
  onClick={[Function]}
>
  
  <span
    className="ExitButton__exitIcon"
  />
</a>
`;

exports[`app/shared-components/ExitButton More Snapshot tests with react-test-renderer, but for edge cases edge case where text is false 1`] = `
<a
  aria-current={null}
  className="ExitButton__exit flex-container align-middle"
  href="/example"
  onClick={[Function]}
>
  <span
    className="ExitButton__exitIcon"
  />
</a>
`;

exports[`app/shared-components/ExitButton Snapshot tests with react-test-renderer exit button is rendered correctly 1`] = `
<a
  aria-current="page"
  className="ExitButton__exit flex-container align-middle active"
  href="/"
  onClick={[Function]}
  style={Object {}}
>
  <span
    className="ExitButton__exitText"
  >
    Exit
  </span>
  <span
    className="ExitButton__exitIcon"
  />
</a>
`;
